src/main/scala/histogramactors/HistogramMain.scala

```diff
@@ -0,0 +1,137 @@
+package histogramactors
+
+import akka.actor.{Actor, ActorRef, ActorSystem, Props}
+import akka.pattern.{ask, pipe}
+import akka.util.Timeout
+
+import java.io.PrintWriter
+import scala.collection.mutable
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.concurrent.duration.{Duration, MILLISECONDS}
+import scala.io.Source
+import scala.util.{Failure, Success, Try, Using}
+
+object HistogramMain extends App {
+  val inputFilePath = "src/main/resources/histogram/input.txt"
+  val outputFilePath = "src/main/resources/histogram/output.txt"
+
+  val system = ActorSystem("histogram")
+  val master = system.actorOf(Props(classOf[Master], inputFilePath, outputFilePath, system), "master")
+  master ! Start
+
+}
+
+class Master(input: String, output: String, system: ActorSystem) extends Actor {
```

Comment:
Try simplifying design:

-  there are too many very fine-grained actors
- overall design is computationally ineffective (I guess messaging overhead is greater than the job itself), try splitting a file into batches, process each batch using the mapper, then combine results at the reducer.

Reply:
Reworked in following way:
1) sequence of lines are grouped to blocks with size = (lines.size / 10) + 1
2) Line case class changed to Block with Seq[String] as input
3) Parser works with whole block of lines now and sends result to reducer